src/MorphoTokenOptimism.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity 0.8.27;
+
+import {IOptimismMintableERC20} from "./interfaces/IOptimismMintableERC20.sol";
+import {IERC165} from
+    "../lib/openzeppelin-contracts-upgradeable/lib/openzeppelin-contracts/contracts/utils/introspection/IERC165.sol";
+
+import {Token} from "./Token.sol";
+
+/// @title MorphoTokenOptimism
+/// @author Morpho Association
+/// @custom:contact security@morpho.org
+/// @notice The Morpho token contract for Optimism networks.
+contract MorphoTokenOptimism is Token, IOptimismMintableERC20 {
+    /* CONSTANTS */
+
+    /// @dev The name of the token.
+    string internal constant NAME = "Morpho Token";
+
+    /// @dev The symbol of the token.
+    string internal constant SYMBOL = "MORPHO";
+
+    // keccak256(abi.encode(uint256(keccak256("morpho.storage.OptimismMintableERC20")) - 1)) & ~bytes32(uint256(0xff))
+    bytes32 internal constant OptimismMintableERC20StorageLocation =
+        0x6fd4c0a11d0843c68c809f0a5f29b102d54bc08a251c384d9ad17600bfa05d00;
+
+    /* STORAGE LAYOUT */
+
+    /// @custom:storage-location erc7201:morpho.storage.OptimismMintableERC20
+    struct OptimismMintableERC20Storage {
+        address _remoteToken;
+        address _bridge;
```

Comment:
couldn't they both be immutable? like in https://docs.optimism.io/builders/app-developers/tutorials/standard-bridge-custom-token

the token is upgradable in case